account_asset_management/wizard/account_asset_compute.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# Copyright 2009-2017 Noviat
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models, _
+
+
+class AccountAssetCompute(models.TransientModel):
+    _name = 'account.asset.compute'
+    _description = "Compute Assets"
+
+    period_id = fields.Many2one(
+        comodel_name='account.period', string='Period', required=True,
+        domain="[('special', '=', False), ('state', '=', 'draft')]",
+        default=lambda self: self._default_period_id(),
+        help="Choose the period for which you want to automatically "
+             "post the depreciation lines of running assets")
+    note = fields.Text()
+
+    @api.model
+    def _default_period_id(self):
+        ctx = dict(self._context, account_period_prefer_normal=True)
+        periods = self.env['account.period'].with_context(ctx).find()
+        periods = periods.filtered(lambda r: r.state == 'draft')
+        if periods:
+            return periods[0]
+        return False
+
+    @api.multi
+    def asset_compute(self):
+        assets = self.env['account.asset'].search(
+            [('state', '=', 'open'), ('type', '=', 'normal')])
+        created_move_ids, error_log = assets._compute_entries(
+            self.period_id, check_triggers=True)
+
+        if error_log:
+            module = __name__.split('addons.')[1].split('.')[0]
+            result_view = self.env.ref(
+                '%s.%s_view_form_result'
+                % (module, self._name))
```

Comment:
@luc-demeyer I believe your self._name results in a xml_id that can't be found. This results in : 'account_asset_management.account.asset.compute_view_form_result' While it should be 'account_asset_management.account_asset_compute_view_form_result'. 
I believe a simple `self._name.replace(".", "_")` should do the trick here.

Reply:
:+1: Or just hardcode the name for better readability.